griptape/artifacts/list_artifact.py

```diff
@@ -9,15 +10,17 @@
 if TYPE_CHECKING:
     from collections.abc import Sequence
 
+T = TypeVar("T", bound=BaseArtifact)
```

Comment:
what is the benefit to the generic type if its bound to `BaseArtifact`?

Reply:
i understand,  but what is the benefit of that vs what is currently there, ie:
```
value: Sequence[BaseArtifact]
```
is that not the same thing?

Reply:
It is not the same thing. If we don't use the generic there it will not catch a scenario like this:

```python
assert ListArtifact[BlobArtifact]([TextArtifact("foo")])
```

Reply:
i see, its for type hinting. makes sense now